node/impl/full/state.go

```diff
@@ -617,14 +617,15 @@ func (m *StateModule) StateWaitMsg(ctx context.Context, msg cid.Cid, confidence
 
 		t, err := stmgr.GetReturnType(ctx, m.StateManager, vmsg.To, vmsg.Method, ts)
 		if err != nil {
-			return nil, xerrors.Errorf("failed to get return type: %w", err)
-		}
-
-		if err := t.UnmarshalCBOR(bytes.NewReader(recpt.Return)); err != nil {
-			return nil, err
+			// This is not nececessary an error -- EVM methods (and in the future native actors) may
+			//  return just bytes.
+			log.Debugf("failed to get return type: %s", err)
```

Comment:
Besides the unknown return type, after this change, we will also ignore errors indicating that loading actor failed or the method is unknown. Are we fine with ignoring them?

Reply:
that's a question for ayush, but it is kind if inevitable when native actors land.

Reply:
Assuming that Wasm actors will exchange CBOR (at least for M2.2, I doubt we’ll want to introduce multicodec shenanigans then), you could make `GetReturnType` return `cbor.Value` when the method cannot be identified. Then the serde should work just fine and we don't need to uglily patch things here.

Reply:
cc @arajasek

Reply:
There is a potentially better solution: introduce a non decoding variant of StateWaitMsg, so that we dont even try to deserialize in there and we use that onr.
Extant code can keep using the strict version.

Reply:
I don't think Lotus needs more API entropy.

Reply:
the current API aint right however, we do want to wait for messages without caring about decoding the return value.

Reply:
Yep, agree. Could you do a downstream usage analysis of the decoded field? I see it's not even being used at the CLI level.

Reply:
Sure, i can grep, but iys a public API so who really knows what users do...

Reply:
so nothing in cli touches ReturnDec; grepping the rest of lotus yields similar results.

Reply:
So to summarize:
- nothing in the entire lotus code base uses `ReturnDec`.
- we can just drop the field if we want and nothing would break.
- current patch (don't cry if it is missing) seems correct and future proof.

Reply:
- We definitely need to keep the existing functionality, there are users depending on it (eg. figuring out the `txn_id` of a `Propose`d multisig message).
- Do not want new method pls.
- I would propose adding an `ErrNotFound` that `GetReturnType` and `GetParamType` can return, and simply returning the raw bytes in that case. I'm not sure if that has a similar effect to @raulk's `cbor.Value` proposition (not sure what `cbor.Value` does).

Reply:
https://github.com/filecoin-project/lotus/pull/9635/commits/e019b206b521166ef053ab20845a10183732ed7b